### ISSUE TITLE
Incomplete Value on JSON/YAML Export for Nested Attributes with Same Name As Top-Level Attribute 

### ISSUE THREAD

**Issue by devstein** [cue-lang/cue#1612]
I assume this is a bug, but please let me know if this is expected behavior! 

### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
cue version v0.4.2 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes

### What did you do?

I tried to export the following Cue to JSON or YAML. 

_Note_: Exporting to Cue works as expected

```cue
#x: {
  y: string
  
  nested: {
    "y": y
  }
}

x: #x & {
  y: "hello world"
}
```

### What did you expect to see?

```yaml
x:
  "y":  "hello world"

  nested:
    "y": "hello world"
```

### What did you see instead?

```shell
failed to encode: x.nested.y: incomplete value _
```

**Comment by seh** [comment]
Introducing an alias helps, per [this playground example](https://tip.cuelang.org/play/?id=x684xrPIvFC#cue@export@yaml).

**Comment by mpvl** [comment]
@seh: the additional alias should not make a difference here. So this is a bug.

However, I cannot reproduce it on the command line. Can you say what you say what you did to produce this bug?

The bug does manifest itself in the playground.

**Comment by devstein** [comment]
@mpvl I first encountered in the playground, then reproduced with an internal build script that uses `yaml.MarshalStream` to generate the output

**Comment by mpvl** [comment]
@devstein: in that case, it would be great if you could post a reproducer here outside of the playground. As it stands it is a bit hard to debug.

**Comment by mvdan** [comment]
I can reproduce with the playground at v0.5.0 (more exactly, it's f681271a38ec, one commit after v0.5.0 on the release branch). See https://cuelang.org/play/?id=SnCOz1sj1Yg#cue@export@json, although this link will change behavior soon, once v0.6.0 is out.

The good news is that the bug appears to be fixed in tip, which will soon become v0.6.0 - https://tip.cuelang.org/play/?id=SnCOz1sj1Yg#cue@export@json (version da3ee7dc7b27) correctly exports both JSON and YAML without an error.
